app_check/src/android/common_android.cc

```diff
@@ -0,0 +1,163 @@
+// limitations under the License.
+
+#include "app_check/src/android/common_android.h"
+
+#include "app/src/app_common.h"
+#include "app/src/util_android.h"
+#include "app_check/src/android/common_android.h"
+#include "app_check/src/include/firebase/app_check.h"
+
+namespace firebase {
+namespace app_check {
+namespace internal {
+
+METHOD_LOOKUP_DEFINITION(app_check_provider,
+                         PROGUARD_KEEP_CLASS
+                         "com/google/firebase/appcheck/AppCheckProvider",
+                         APP_CHECK_PROVIDER_METHODS)
+
+// Used to setup the cache of AppCheckProvider interface method IDs to reduce
+// time spent looking up methods by string.
+// clang-format off
+#define APP_CHECK_TOKEN_METHODS(X)                                                        \
+  X(GetToken, "getToken",                                                          \
+    "()Ljava/lang/String;"),                                                        \
+  X(GetExpireTimeMillis, "getExpireTimeMillis",                                                          \
+    "()J")
+// clang-format on
+
+METHOD_LOOKUP_DECLARATION(app_check_token, APP_CHECK_TOKEN_METHODS)
+METHOD_LOOKUP_DEFINITION(app_check_token,
+                         PROGUARD_KEEP_CLASS
+                         "com/google/firebase/appcheck/AppCheckToken",
+                         APP_CHECK_TOKEN_METHODS)
+
+static const char* kApiIdentifier = "AppCheckProvider";
+
+bool CacheCommonAndroidMethodIds(JNIEnv* env, jobject activity) {
+  // Cache the token and provider classes.
+  if (!(app_check_token::CacheMethodIds(env, activity) &&
+        app_check_provider::CacheMethodIds(env, activity))) {
+    return false;
+  }
+
+  return true;
+}
+
+void ReleaseCommonAndroidClasses(JNIEnv* env) {
+  app_check_token::ReleaseClass(env);
+  app_check_provider::ReleaseClass(env);
+}
+
+// Converts an android AppCheckToken to a C++ AppCheckToken
+AppCheckToken CppTokenFromAndroidToken(JNIEnv* env, jobject token_obj) {
+  AppCheckToken cpp_token;
+
+  if (token_obj != nullptr) {
+    jobject token_str = env->CallObjectMethod(
+        token_obj, app_check_token::GetMethodId(app_check_token::kGetToken));
+    util::CheckAndClearJniExceptions(env);
+    cpp_token.token = util::JStringToString(env, token_str);
+
+    jlong millis = env->CallLongMethod(
+        token_obj,
+        app_check_token::GetMethodId(app_check_token::kGetExpireTimeMillis));
+    util::CheckAndClearJniExceptions(env);
+    cpp_token.expire_time_millis = static_cast<int64_t>(millis);
+  }
+
+  return cpp_token;
+}
+
+// TODO: is this a good approach? can I get JNI env before I have initialized an
+// app? I found this method in credential android
+// Alternatively I can have a non-static method that uses app_
+JNIEnv* GetJniEnv() {
+  // The JNI environment is the same regardless of App.
+  App* app = app_common::GetAnyApp();
+  FIREBASE_ASSERT(app != nullptr);
+  return app->GetJNIEnv();
+}
+
+namespace {
```

Comment:
I've seen this anonymous namespace pattern used in a couple places to contain code that is only relevant to the current file.

In this case, I did it because I was looking at this example: https://github.com/firebase/firebase-cpp-sdk/blob/main/storage/src/android/storage_reference_android.cc#L193

I don't feel strongly about the anonymous namespace though, and it could be removed if you prefer.